app/views/layouts/_sidebar_nav.html.erb

```diff
@@ -10,15 +11,15 @@
     </a>
   <% end %>
   <div class="sidebar-navigation-links">
-    <% NavigationLink.ordered[0..4].each do |link| %>
+    <% nav_links.ordered[0..4].each do |link| %>
       <%= render "articles/sidebar_nav_link", link: link %>
     <% end %>
 
-    <% if NavigationLink.all.length > 5 %>
+    <% if nav_links.all.length > 5 %>
```

Comment:
You can remove `.all` from here and in the partial below since we are doing that above now :) 

Also, I just learned that `.all` loads an AR relation which means we still hit the db 3 times here. In order to prevent that
```ruby
nav_links = NavigationLink.all.to_a
```

ok, I think you should be good now with that! 🙈  learning all the time!

Reply:
<3